ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/common/RecoveryEnvironmentBean.java

```diff
@@ -90,6 +91,11 @@ public int getPeriodicRecoveryPeriod()
     public void setPeriodicRecoveryPeriod(int periodicRecoveryPeriod)
     {
         this.periodicRecoveryPeriod = periodicRecoveryPeriod;
+
+        if (this.periodicRecoveryPeriod != PeriodicRecovery._defaultRecoveryPeriod &&  tsLogger.logger.isDebugEnabled()) {
```

Comment:
Remark: I would check if debug is enabled before performing these conditional tests since it it performs better and is more logical (the purpose of the `if statement` is to log a debug message so logically the log level needs to be checked first).